### PR TITLE
Consolidate CDC env file.

### PR DESCRIPTION
**Pull request by keyurva**
* Consolidates CDC env vars into a single file (`env.list`).
* See [go/cdc-env-file-consolidation](http://go/cdc-env-file-consolidation) for more details.
* As part of the consolidation `sqlite_env.list` and `cloudsql_env.list` need to be removed.
  + We'll remove them from the code once @kmoscoe has removed their mention from the docs.

* `Dockerfile` changes:
  + `lancedb` deps were moved to a different requirements file a few weeks ago. Adapting Dockerfile to that move.
  + Uses `npm ci` vs `npm install` for faster builds. See [this thread](https://stackoverflow.com/a/74068335) for more details.
* The Dockerfile changes are not related to env consolidation so I can do it in a separate PR if needed. LMK.
* Built docker image (`gcr.io/datcom-ci/datacommons-website-compose:keyurs-test`) with these changes and verified in the cloud [here](https://screenshot.googleplex.com/AdiWt7P4umvBuX4).

### CODE REVIEW

File: custom_dc/env.list [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
 DC_API_KEY=
 # API key for accessing Maps and Places APIs for exploration tools.
 MAPS_API_KEY=
+
+### Directories ###
+
+# Input directory in Google Cloud Storage or local drive, containing files to be processed.
+INPUT_DIR=
+# Output directory in Google Cloud Storage or local drive for generated csv files.
+# Leave blank if same as INPUT_DIR.
```
Comment (hqpho) [inline custom_dc/env.list:13]: could maybe mark this as an optional param?

File: build/web_compose/compose.sh [hunk 2]
```diff
@@ -19,21 +19,47 @@ export TOKENIZERS_PARALLELISM=false
...
+if [[ $OUTPUT_DIR == "" && $GCS_DATA_PATH != "" ]]; then
+    echo "GCS Data Path: $GCS_DATA_PATH"
+    echo "GCS_DATA_PATH is deprecated and will be removed in the future. Use OUTPUT_DIR instead."
+    export OUTPUT_DIR=$GCS_DATA_PATH
 fi
 
-nginx -c /workspace/nginx.conf
+# Set OUTPUT_DIR to same as INPUT_DIR if not specified.
+if [[ $OUTPUT_DIR == "" ]]; then
+    if [[ $INPUT_DIR == "" ]]; then
```
Comment (hqpho) [inline build/web_compose/compose.sh:31]: Should we do this check even if OUTPUT_DIR is set?

**Review by hqpho — APPROVED** [review]

File: custom_dc/env.list [hunk 3]
```diff
@@ -1,7 +1,20 @@
+### API Keys ###
```
Comment (kmoscoe) [inline custom_dc/env.list:1]: Keys > keys

File: custom_dc/env.list [hunk 4]
```diff
@@ -1,7 +1,20 @@
...
+
 # API key for accessing the base Data Commons API.
 DC_API_KEY=
 # API key for accessing Maps and Places APIs for exploration tools.
 MAPS_API_KEY=
+
+### Directories ###
+
+# Input directory in Google Cloud Storage or local drive, containing files to be processed.
```
Comment (kmoscoe) [inline custom_dc/env.list:10]: Please add a line about how to specify the directory format for GCS, i.e. gs://<GCS bucket>/<directory>

File: build/web_compose/compose.sh (hunk 2 above)
Comment (keyurva) [reply build/web_compose/compose.sh:31]: Probably not. This docker ("services docker") is about serving data from the output directory so if it is set, that's the one we want to use.

File: custom_dc/env.list (hunk 3 above)
Comment (keyurva) [reply custom_dc/env.list]: Done.

File: custom_dc/env.list (hunk 4 above)
Comment (keyurva) [reply custom_dc/env.list]: Done.

File: custom_dc/env.list (hunk 1 above)
Comment (keyurva) [reply custom_dc/env.list:15]: This one is tricky. This is actually one of the most important variables for the services docker.

Ideally, I would've liked to do `OUTPUT_DIR=$INPUT_DIR` but docker does not apply substitution on these variables. ([ref](https://stackoverflow.com/a/45923906))

I'm wondering if it's best to make it required instead and if both are the same, require that both be set to the same value.

Thoughts?

Comment (keyurva) [reply custom_dc/env.list:15]: Merging it as-is for now. Can do a follow-up if we decide to make it required.

**Review by keyurva — COMMENTED** [review]
Thanks for the reviews.

**Pull request merged**